zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -95,10 +95,10 @@ sealed trait HttpCodec[-AtomTypes, Value] {
   final def &[Value2](
     that: QueryCodec[Value2],
   )(implicit
-    combiner: Combiner[Value, Value2],
+    combiner: Combiner[NonEmptyChunk[Value], NonEmptyChunk[Value2]],
```

Comment:
I think it's risky to change this combiner since you don't know what the left hand side is. Possibly you should not change `QueryCodec` so that the `NonEmptyChunk` is already reflected in the type parameters and you don't need to make any changes here.